File: sdks/go/inventory.go
package puxbay

import "fmt"

// InventoryService handles inventory-related API calls
type InventoryService struct {
	client *Client
}

// StockLevel represents current stock information
type StockLevel struct {
	ProductID string `json:"product_id"`
	Quantity  int    `json:"quantity"`
	Branch    string `json:"branch"`
}

// GetStockLevels retrieves current stock levels
func (s *InventoryService) GetStockLevels(branchID string) ([]StockLevel, error) {
	endpoint := "inventory/stock-levels/"
	if branchID != "" {
		endpoint = fmt.Sprintf("%s?branch=%s", endpoint, branchID)
	}

	var result []StockLevel
	err := s.client.GET(endpoint, &result)
	return result, err
}

// GetLowStock retrieves products with low stock
func (s *InventoryService) GetLowStock(threshold int) ([]Product, error) {
	endpoint := fmt.Sprintf("inventory/low-stock/?threshold=%d", threshold)
	var result []Product
	err := s.client.GET(endpoint, &result)
	return result, err
}

// CreateTransfer creates a new stock transfer
func (s *InventoryService) CreateTransfer(transfer *StockTransfer) (*StockTransfer, error) {
	var result StockTransfer
	err := s.client.POST("stock-transfers/", transfer, &result)
	return &result, err
}

// ListTransfers retrieves a list of stock transfers
func (s *InventoryService) ListTransfers(page int, status string) (*PaginatedResponse, error) {
	endpoint := fmt.Sprintf("stock-transfers/?page=%d", page)
	if status != "" {
		endpoint = fmt.Sprintf("%s&status=%s", endpoint, status)
	}

	var result PaginatedResponse
	err := s.client.GET(endpoint, &result)
	return &result, err
}

File: sdks/python/puxbay/resources/inventory.py
"""
Inventory resource client
"""

from typing import Dict, Any, Optional


class Inventory:
    """Client for inventory-related API endpoints"""
    
    def __init__(self, client):
        self.client = client
    
    def get_stock_levels(self, branch_id: Optional[str] = None) -> Dict[str, Any]:
        """
        Get current stock levels for all products
        
        Args:
            branch_id: Filter by specific branch (optional)
        
        Returns:
            List of products with current stock quantities
        """
        params = {}
        if branch_id:
            params['branch'] = branch_id
        return self.client.get('inventory/stock-levels/', params=params)
    
    def get_low_stock(self, threshold: int = 10) -> Dict[str, Any]:
        """
        Get products with low stock
        
        Args:
            threshold: Stock quantity threshold (default: 10)
        
        Returns:
            List of products below the threshold
        """
        return self.client.get('inventory/low-stock/', params={'threshold': threshold})
    
    def create_transfer(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Create a stock transfer between branches
        
        Args:
            data: Transfer data (from_branch, to_branch, items)
        
        Returns:
            Created transfer record
        
        Example:
            >>> transfer_data = {
            ...     "from_branch": "branch-uuid-1",
            ...     "to_branch": "branch-uuid-2",
            ...     "items": [
            ...         {"product": "product-uuid", "quantity": 10}
            ...     ]
            ... }
            >>> transfer = client.inventory.create_transfer(transfer_data)
        """
        return self.client.post('stock-transfers/', json=data)
    
    def list_transfers(self, page: int = 1, status: Optional[str] = None) -> Dict[str, Any]:
        """
        List stock transfers
        
        Args:
            page: Page number (default: 1)
            status: Filter by status (pending, completed, cancelled)
        
        Returns:
            Paginated list of transfers
        """
        params = {'page': page}
        if status:
            params['status'] = status
        return self.client.get('stock-transfers/', params=params)
    
    def create_stocktake(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Create a stocktake (physical inventory count)
        
        Args:
            data: Stocktake data (branch, items with counted quantities)
        
        Returns:
            Created stocktake record
        """
        return self.client.post('stocktakes/', json=data)
    
    def list_stocktakes(self, page: int = 1) -> Dict[str, Any]:
        """
        List stocktakes
        
        Args:
            page: Page number (default: 1)
        
        Returns:
            Paginated list of stocktakes
        """
        return self.client.get('stocktakes/', params={'page': page})
